typescript/cli/src/commands/warp.ts

```diff
@@ -415,68 +416,84 @@ export const check: CommandModuleWithContext<{
 };
 
 export const rebalancer: CommandModuleWithWriteContext<{
-  warpRouteId: string;
-  checkFrequency: number;
-  rebalancerConfigFile: string;
+  configFile: string;
+  warpRouteId?: string;
+  checkFrequency?: number;
   withMetrics?: boolean;
+  monitorOnly?: boolean;
 }> = {
   command: 'rebalancer',
   describe: 'Run a warp route collateral rebalancer',
   builder: {
+    configFile: {
+      type: 'string',
+      description:
+        'The path to a rebalancer configuration file (.json or .yaml)',
+      demandOption: true,
+      alias: ['rebalancerConfigFile'],
+    },
     warpRouteId: {
       type: 'string',
       description: 'The warp route ID to rebalance',
-      demandOption: true,
+      demandOption: false,
     },
     checkFrequency: {
       type: 'number',
       description: 'Frequency to check balances in ms',
-      demandOption: true,
-      alias: 'v',
-    },
-    rebalancerConfigFile: {
-      type: 'string',
-      description:
-        'The path to a rebalancer configuration file (.json or .yaml)',
-      demandOption: true,
-      alias: 's',
+      demandOption: false,
     },
     withMetrics: {
       type: 'boolean',
       description: 'Enable metrics',
       demandOption: false,
-      alias: 'm',
+    },
+    monitorOnly: {
+      type: 'boolean',
+      description: 'Run in monitor only mode',
+      demandOption: false,
     },
   },
   handler: async ({
     context,
+    configFile,
     warpRouteId,
     checkFrequency,
-    rebalancerConfigFile,
-    withMetrics = false,
+    withMetrics,
+    monitorOnly,
   }) => {
     try {
+      const { registry, key: rebalancerKey } = context;
+
+      // This might happen because despite the key being typed as required, it is not validated beforehand.
+      if (!rebalancerKey) {
+        throw new Error('No key found in context');
+      }
```

Comment:
shouldn't we attempt to prompt for the key if the command is not running in a CI context?

Reply:
Fixed at fbf17e8b784fd8be69d3f9515b2fad625697b575